src/Symfony/Component/Messenger/Transport/Receiver/ReceiverInterface.php

```diff
@@ -21,13 +24,29 @@ interface ReceiverInterface
     /**
      * Receive some messages to the given handler.
      *
-     * The handler will have, as argument, the received {@link \Symfony\Component\Messenger\Envelope} containing the message.
+     * The handler will have, as argument, the received {@link \Symfony\Component\Messenger\Envelope}.
      * Note that this envelope can be `null` if the timeout to receive something has expired.
+     *
+     * If the received message cannot be decoded, the message should
+     * be removed from the queue and the MessageDecodingFailedException be thrown
+     *
+     * @throws TransportException If there is an issue communicating with the transport
      */
     public function receive(callable $handler): void;
 
     /**
      * Stop receiving some messages.
      */
     public function stop(): void;
+
+    /**
+     * Return if the receiver should respond/ack to a queue.
+     *
+     * If this receiver does not communicate back to a queue,
+     * this can return null. In that case, retry will not
+     * be attempted and messages will not be acknowledged or
+     * rejected on the responder when inside the receive handler
+     * callback.
+     */
+    public function getResponder(): ?ReceiverResponderInterface;
```

Comment:
This is only used in the `Worker`. Instead, can you require to be injected the `TransportInterface`, get the sender and check whether it implements `ReceiverResponderInterface`? This way, we don't break this interface (which is implemented by a lot of transports already) as we don't need.

Reply:
I like your thinking. Oh, bit did you mean "get the sender" or "get the receiver" above? I want to make sure I wasn't misunderstanding that part.

However, as much as I like this, there are two problems:

1) If we inject the `TransportInterface` into `Worker` instead of the `ReceiverInterface`, the decorator classes (like `StopWhenTimeLimitIsReachedReceiver`) would need to be updated to decorate the whole transport).

2) The `TransportInterface` does not have a `getReceiver()` method. So, if we inject the transport into `Worker`, we can't get the receiver without adding this method to the interface (which would still break existing implementations).

Thoughts?

Reply:
After talking with Nicolas, I've updated this (again). The "responder" is gone (yay!), `ReceiverInterface` has `acknowledgeMessage()` and `rejectMessage()` (because any "decent" receivers will have this, others can leave them blank) and for "retry", we're actually sending the Envelope back through the bus in order to re-send (Nicolas' stroke of genius). We simply remove the `ReceivedStamp` and pass it back in.